src/api/use-subscription.ts

```diff
@@ -87,19 +81,40 @@ export function useSubscription({
         webSocket.close();
       }
     };
```

Comment:
`webSocket` state variable is used in the clean up function, but it isn't in the dependency list, as it should be when [it is a reactive value](https://react.dev/learn/removing-effect-dependencies#to-remove-a-dependency-prove-that-its-not-a-dependency). However adding it to the dependency list would give a loop. 

One way to solve this is by also have the `webSocket` as a ref instead of a state value. Another option is to utilise the previous state in the `setWebSocket` function to retrieve the `webSocket` value:
```ts
// Cleanup
return () => {
  if (retryTimeout) clearTimeout(retryTimeout);
  setWebSocket(ws => {
    if (ws) {
      ws.onclose = null;
      onClose && onClose({code: 1000, reason: 'Cleanup'});
      ws.close();
    }
    return undefined;
  }
};
```

I am really focusing on the usage of `useEffect`s in PRs lately as I think missing dependencies and dependency loops have been, and still are, the cause of some hard-to-debug errors in the app. I have a goal of enabling the exhaustive deps linter rule to catch these situations.

Reply:
Aha, updated now. Sounds like a good idea to enable the rule 👍